src/ray/gcs/store_client/in_memory_store_client.cc

```diff
@@ -127,9 +129,7 @@ Status InMemoryStoreClient::AsyncBatchDelete(const std::string &table_name,
 }
 
 int InMemoryStoreClient::GetNextJobID() {
-  absl::MutexLock lock(&mutex_);
-  job_id_ += 1;
-  return job_id_;
+  return job_id_.fetch_add(1, std::memory_order_acq_rel);
```

Comment:
`fetch_add` returns the value before the addition operation, which is not the same current behavior, is that intended?

Reply:
correct, i updated job_id initialization to 1 now though so it should keep the same behavior

Reply:
also went with acq_rel bc 100% sure it's safe but i think memory_order_relaxed should also work in this situation

Reply:
For single thread use case, there's no reorder operation, so relaxed memory order is good enough